proof/refine/ARM/TcbAcc_R.thy

```diff
@@ -2413,15 +2426,21 @@ lemma rescheduleRequired_corres:
                    split: option.splits Structures_A.kernel_object.splits)
 
   apply (rule corres_symb_exec_l[OF _ _ assert_sp, rotated])
-    apply (clarsimp simp: exs_valid_def assert_def return_def obj_at_def weak_valid_sched_action_def
-                          vs_all_heap_simps)
-   apply (clarsimp simp: no_fail_def  return_def obj_at_def weak_valid_sched_action_def
-                         vs_all_heap_simps)
-
+    apply (clarsimp simp: exs_valid_def return_def vs_all_heap_simps
+                          obj_at_def pred_tcb_at_def weaker_valid_sched_action_def)
+   apply (clarsimp simp: no_fail_def return_def vs_all_heap_simps
+                         obj_at_def pred_tcb_at_def weaker_valid_sched_action_def)
   apply (corressimp corres: tcbSchedEnqueue_corres
                       simp: obj_at_def is_tcb_def weak_sch_act_wf_def)
   done
 
+lemma rescheduleRequired_corres:
```

Comment:
Nice! I'm surprised this didn't already exist.

Reply:
It actually did, but I needed to weaken the precondition because `weak_valid_sched_action` can't be propagated backwards over the `thread_set` instance in `sched_context_donate`. Hence, I've introduced a weaker variant of the corres rule to use in this particular case and moved the proofs to that rule to avoid boilerplating.